text/reader.go

```diff
@@ -45,14 +45,14 @@ type Reader interface {
 	SetPosition(int, Segment)
 
 	// SetPadding sets padding to the reader.
-	SetPadding(int)
+	SetPadding(int, []byte)
 
 	// Advance advances the internal pointer.
 	Advance(int)
 
 	// AdvanceAndSetPadding advances the internal pointer and add padding to the
 	// reader.
-	AdvanceAndSetPadding(int, int)
+	AdvanceAndSetPadding(int, int, []byte)
```

Comment:
Hm. I wonder if this makes sense. The `int` is now not needed if we have full padding bytes, so maybe let's just pass full padding then? 

Maybe we can abstract tabs and spaces as well, no need to send actual bytes if we have e.g `4 tabs, 5 spaces, 1 tab` padding (: 

Same with `SetPadding`

Reply:
No it is still needed. The padding can be `1`, `2`, `3`, `4` while there is still just one tab. We do not have enough information, it's not enough to have just tab

Example

```
*.foo
..```Markdown
..[tb].foo
..```
```

padding is 2, while chars is `\t`

<code>
```Markdown
[tb].foo
```
</code>

padding is 4, while chars is `\t`

you cannot tell padding just by looking at the characters

Reply:
Or - according to the commonmark, this
```
-[t][tb]text
```

needs to parse as "code block text with two spaces in front of it"

so only the second half of `[tb]` is the padding

...it is quite confusing at first. Markdown is hard :D

Reply:
right. 

Thanks for explaining. 🤗 

I wonder what maintainers think!

Markdown is not easy. I thought distributed systems are hard but Markdown formatting and parsing is much harder .... Kudos to all people who allow us to forget about this 🤗